outline/shadowsocks/connectivity.go

```diff
@@ -32,12 +32,7 @@ const reachabilityTimeout = 10 * time.Second
 // the current network. Parallelizes the execution of TCP and UDP checks, selects the appropriate
 // error code to return accounting for transient network failures.
 // Returns an error if an unexpected error ocurrs.
-func CheckConnectivity(host string, port int, password, cipher string) (int, error) {
-	client, err := shadowsocks.NewClient(host, port, password, cipher)
-	if err != nil {
-		// TODO: Inspect error for invalid cipher error or proxy host resolution failure.
-		return Unexpected, err
-	}
+func CheckConnectivity(host string, port int, client shadowsocks.Client) (int, error) {
```

Comment:
Is this only used by Electron, not Cordova?

Reply:
You're right, this is used by Cordova, and `shadowsocks.Client` is presumably not gobind-compatible, so I don't think we can export it here.  I've reorganized this PR to deal with the gobind restrictions using a Config struct and a client wrapper struct.